pkg/net/http/http.go

```diff
@@ -46,3 +49,17 @@ func PickHeader(header http.Header, key, defaultValue string) string {
 
 	return defaultValue
 }
+
+// PickTypedHeader pick typed header with key.
+func PickTypedHeader[T any](header http.Header, key string, defaultValue T) T {
```

Comment:
Refactor `PickHeader` and remove `PickTypedHeader`.

Reply:
PickTypedHeader has been removed